bee-ternary/src/t3b1.rs

```diff
@@ -1,28 +1,56 @@
 use std::ops::Range;
-use crate::{Trit, RawEncoding, RawEncodingBuf};
+use crate::{
+    Btrit,
+    Utrit,
+    RawEncoding,
+    RawEncodingBuf,
+    ShiftTernary
+};
 
-const BASE: usize = 3;
+const TPB: usize = 3;
+const BAL: i8 = 13;
 
 #[repr(transparent)]
 pub struct T3B1([()]);
 
 impl T3B1 {
-    unsafe fn make(ptr: *const u8, offset: usize, len: usize) -> *const Self {
-        let len = (len << 2) | (offset % BASE);
-        std::mem::transmute((ptr.offset((offset / BASE) as isize), len))
+    pub(crate) unsafe fn make(ptr: *const i8, offset: usize, len: usize) -> *const Self {
+        let len = (len << 2) | (offset % TPB);
+        std::mem::transmute((ptr.offset((offset / TPB) as isize), len))
     }
 
-    unsafe fn ptr(&self, index: usize) -> *const u8 {
-        let byte_offset = (self.len_offset().1 + index) / BASE;
-        (self.0.as_ptr() as *const u8).offset(byte_offset as isize)
+    unsafe fn ptr(&self, index: usize) -> *const i8 {
+        let byte_offset = (self.len_offset().1 + index) / TPB;
+        (self.0.as_ptr() as *const i8).offset(byte_offset as isize)
     }
 
     fn len_offset(&self) -> (usize, usize) {
         (self.0.len() >> 2, self.0.len() & 0b11)
     }
 }
 
+fn extract(x: i8, elem: usize) -> Btrit {
+    if elem < TPB {
+        Utrit::from_u8((((x + BAL) / 3i8.pow(elem as u32)) % 3) as u8).shift()
+    } else {
+        unreachable!("Attempted to extract invalid element {} from balanced T3B1", elem)
+    }
+}
+
+fn insert(x: i8, elem: usize, trit: Btrit) -> i8 {
```

Comment:
Maybe i'm not understanding the intention of this method, but it seems that it inserts a trit at some index and returns the new byte value so it can be used,

so if i call:
`insert(0,0,1)`
it should return 1 as i8
calling it again like that:
`insert(1,0,0)` should return `0` imo since it's redoing the previous call, right?
but it return `1` and does not change the byte, is this intended? and if so, what am i not understanding?

Reply:
my bad, it works fine